### PR TITLE
E05XX odyssey

### PR DESCRIPTION
**Pull request by zackmdavis**
chipping away at the surface exposed by #43709

r? @GuillaumeGomez

### CODE REVIEW

File: src/librustc_passes/diagnostics.rs [hunk 1]
```diff
@@ -223,12 +223,43 @@ To fix this, add a label specifying which loop is being broken out of:
 ```
 'foo: while break 'foo {}
 ```
+"##,
+
+E0571: r##"
+The `break` statement can take an argument (which will be the value of the loop
```
Comment (GuillaumeGomez) [inline src/librustc_passes/diagnostics.rs]: A long error explanation is supposed to start by a short summary of the error. The error explanation comes just after the erroneous code example and just before the fix.

File: src/libsyntax/diagnostic_list.rs [hunk 2]
```diff
@@ -162,6 +162,63 @@ For more information about the cfg attribute, read:
 https://doc.rust-lang.org/reference.html#conditional-compilation
 "##,
 
+E0552: r##"
+A unrecognized representation attribute was used.
+
+You can use a `repr` attribute to tell the compiler how you want a struct or
```
Comment (GuillaumeGomez) [inline src/libsyntax/diagnostic_list.rs]: The explanation is supposed to be after the erroneous code example.

File: src/libsyntax/diagnostic_list.rs [hunk 3]
```diff
@@ -162,6 +162,63 @@ For more information about the cfg attribute, read:
...
+["Alternative Representations" section]: https://doc.rust-lang.org/nomicon/other-reprs.html
+"##,
+
+E0554: r##"
+Feature attributes are only allowed on the nightly release channel. Stable or
+beta compilers will not comply.
+
+Example of erroneous code (on a stable compiler):
+
+```ignore (depends on release channel)
```
Comment (GuillaumeGomez) [inline src/libsyntax/diagnostic_list.rs]: Just `ignore` please.

File: src/libsyntax/diagnostic_list.rs [hunk 4]
```diff
@@ -162,6 +162,63 @@ For more information about the cfg attribute, read:
...
+#![feature(non_ascii_idents)] // error: #![feature] may not be used on the
+                              //        stable release channel
+```
+
+If you need the feature, make sure to use a nightly release of the compiler
+(but be warned that the feature may be removed or altered in the future).
+"##,
+
+E0557: r##"
+A feature attribute for a removed feature was used.
```
Comment (GuillaumeGomez) [inline src/libsyntax/diagnostic_list.rs]: Seems a bit complicated when formulated this way.

File: src/libsyntax/feature_gate.rs [hunk 5]
```diff
@@ -1602,7 +1602,7 @@ fn maybe_stage_features(span_handler: &Handler, krate: &ast::Crate,
             if attr.check_name("feature") {
                 let release_channel = option_env!("CFG_RELEASE_CHANNEL").unwrap_or("(unknown)");
                 span_err!(span_handler, attr.span, E0554,
-                          "#[feature] may not be used on the {} release channel",
+                          "#![feature] may not be used on the {} release channel",
```
Comment (GuillaumeGomez) [inline src/libsyntax/feature_gate.rs]: Why this change?

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]
A few things to change but globally it's good.

File: src/libsyntax/diagnostic_list.rs (hunk 3 above)
Comment (kennytm) [inline src/libsyntax/diagnostic_list.rs]: @GuillaumeGomez  Just writing `ignore` without explanation will cause `tidy` failure. `--explain` will strip them out though.

Comment (GuillaumeGomez) [inline src/libsyntax/diagnostic_list.rs]: Oh that's strange.

**Comment by zackmdavis** [comment]
(updated)

File: src/libsyntax/feature_gate.rs (hunk 5 above)
Comment (zackmdavis) [inline src/libsyntax/feature_gate.rs]: See [the commit message](https://github.com/rust-lang/rust/commit/6fa140b8).

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+

**Pull request merged**